indexer-app/src/cmr/indexer/services/index_set_service.clj

```diff
@@ -68,6 +71,38 @@
          (for [index-name index-names-array]
            {(keyword index-name) (gen-valid-index-name prefix-id index-name)})))
 
+(defn get-canonical-key-name
+  "Removes the leading number prefix (e.g., '1_') and trailing shard count (e.g., '_100_shards')
+     from an index name. If the index name represents a concept ID (starts with a letter followed
+     by numbers), formats it as a concept ID with hyphen separator.
+     Examples:
+     '1_small_collections_100_shards' -> 'small_collections'
+     '1_c2317033465_nsidc_ecs' -> 'C2317033465-NSIDC_ECS'
+     '1_collections_v2' -> 'collections-v2'
+     '1_v123_5w5_nsidc_ecs' -> 'V123_5W5-NSIDC_ECS'"
```

Comment:
Code Rabbit found this first, but this isn't a valid index name. We don't put an underscore in the middle of our concept IDs and also this example doesn't work as code rabbit points out.